lib/posix/clock.c

```diff
@@ -34,6 +38,12 @@ int clock_gettime(clockid_t clock_id, struct timespec *ts)
 		break;
 
 	case CLOCK_REALTIME:
+#ifdef CONFIG_RTC
+		dev_rtc = device_get_binding(DT_RTC_0_NAME);
```

Comment:
it's going to be heavy, since it will lookup for the device each and every time clock_gettime() will be called.
Store dev_rtc statically in this file, and if NULL, then run device_get_binding()

Reply:
You're right I fix it!

Reply:
When I store dev_rtc statically, I have some troubles when clock_gettime is called in userspace. I try to find a solution

Reply:
It's because you did not provide the drivers/rtc/rtc_handlers.c which should implement the userland "gateway" functions. See drivers/spi/spi_handlers.c for instance

Reply:
I did it, but when clock_gettime is called, I have this error when my static dev_rtc is read
[00:00:00.005,000] <err> os: ***** MPU FAULT *****
[00:00:00.005,000] <err> os:   Data Access Violation
[00:00:00.005,000] <err> os:   MMFAR Address: 0x20000610
[00:00:00.005,000] <err> os: r0/a1:  0x00000001  r1/a2:  0x20001fc8  r2/a3:  0x00000000
[00:00:00.005,000] <err> os: r3/a4:  0x20000610 r12/ip:  0x08000b7b r14/lr:  0x08000bd7
[00:00:00.005,000] <err> os:  xpsr:  0x61000000
[00:00:00.005,000] <err> os: Faulting instruction address (r15/pc): 0x08008168
[00:00:00.005,000] <err> os: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:00:00.005,000] <err> os: Current thread: 0x20000704 (unknown)
[00:00:00.078,000] <err> os: Halting system

Reply:
I missed the addition, in which commit is it?

Reply:
I push it!

Reply:
I resolved this by turn Turn clock_[get/set]time into system calls